packages/svelte/src/internal/client/runtime.js

```diff
@@ -1273,24 +1282,15 @@ export function push_destroy_fn(signal, destroy_fn) {
 	}
 }
 
+const STATUS_MASK = ~(DIRTY | MAYBE_DIRTY | CLEAN);
 /**
  * @template V
  * @param {import('./types.js').Signal<V>} signal
  * @param {number} status
  * @returns {void}
  */
 export function set_signal_status(signal, status) {
-	const flags = signal.f;
-	if ((flags & status) === 0) {
-		if ((flags & MAYBE_DIRTY) !== 0) {
-			signal.f ^= MAYBE_DIRTY;
-		} else if ((flags & CLEAN) !== 0) {
-			signal.f ^= CLEAN;
-		} else if ((flags & DIRTY) !== 0) {
-			signal.f ^= DIRTY;
-		}
-		signal.f ^= status;
-	}
+	signal.f = (signal.f & STATUS_MASK) | status;
```

Comment:
This reduces both the code size and the bytecode: https://javascript.godbolt.org/z/xTrY7KTe4

I can't read the bytecode lol so verify that it doesn't affect perf negatively

Reply:
Nice